### PR TITLE
Add eager for keras benchmark

### PR DESCRIPTION
**Pull request by yhliang2018**
Hi All,

Could you help to review this PR? This PR is created to refine the keras model benchmark by:
1) Use models_helper for synthetic dataset generation
2) Add eager mode for keras model. 

Some notes about eager mode: to run with eager mode, the optimizer should be TF native; Also, currently eager mode and keras multi_gpu_model are not compatible. With eager mode enabled, only one GPU is used even multiple GPUs are provided.

Thanks,
Yanhui

### CODE REVIEW

File: official/keras_application_models/benchmark_main.py [hunk 1]
```diff
@@ -54,6 +58,10 @@ def run_keras_model_benchmark(_):
     raise AssertionError("The --model command line argument should "
                          "be a key in the `MODELS` dictionary.")
 
+  # Check if eager mode is enabled
```
Comment (karmel) [inline official/keras_application_models/benchmark_main.py]: Can we add the caveats you mentioned in the PR message to the comments somewhere too? Maybe to the eager flag help.

File: official/keras_application_models/benchmark_main.py [hunk 2]
```diff
@@ -46,6 +46,10 @@
     # "nasnetmobile": tf.keras.applications.NASNetMobile,
 }
 
+# Number of synthetic images as the same number of images in ImageNet
+_IMAGENET_TRAIN_IMAGES = 1281167
+_IMAGENET_EVAL_IMAGES = 50000
```
Comment (karmel) [inline official/keras_application_models/benchmark_main.py]: I wouldn't bother making this complicated for synthetic data. We can re-evaluate if we add real data processing, but for now, maybe a single flag that defaults to 1000?

File: official/keras_application_models/benchmark_main.py [hunk 3]
```diff
@@ -64,30 +71,31 @@ def run_keras_model_benchmark(_):
...
+        FLAGS.model, FLAGS.batch_size)
     val_dataset = dataset.generate_synthetic_input_dataset(
-        FLAGS.model, val_num_images)
+        FLAGS.model, FLAGS.batch_size)
   else:
     raise ValueError("Only synthetic dataset is supported!")
 
   # If run with multiple GPUs
+  # If eager mode is enabled, only one GPU is utilized even multiple GPUs
+  # are provided
```
Comment (anj-s) [inline official/keras_application_models/benchmark_main.py]: Do you want to raise a warning or error message if eager execution is enabled and we use multi_gpu_model?

File: official/keras_application_models/benchmark_main.py [hunk 4]
```diff
@@ -146,6 +154,18 @@ def define_keras_benchmark_flags():
...
+      name="num_images", default=_NUM_IMAGES,
+      help=flags_core.help_wrap(
+          "The number of synthetic images for training and evaluation. The "
+          "default value is 1000."))
+
+  flags.DEFINE_boolean(
+      name="eager", default=False, help=flags_core.help_wrap(
+          "To enable eager execution. Note that if eager execution is enabled, "
+          "only one GPU is utilized even multiple GPUs are provided and "
+          "multi_gpu_model is used."))
```
Comment (anj-s) [inline official/keras_application_models/benchmark_main.py]: s/even/even if

File: official/keras_application_models/benchmark_main.py [hunk 5]
```diff
@@ -146,6 +154,18 @@ def define_keras_benchmark_flags():
       help=flags_core.help_wrap(
           "Model to be benchmarked."))
 
+  flags.DEFINE_integer(
+      name="num_images", default=_NUM_IMAGES,
+      help=flags_core.help_wrap(
+          "The number of synthetic images for training and evaluation. The "
+          "default value is 1000."))
```
Comment (anj-s) [inline official/keras_application_models/benchmark_main.py]: Maybe I missed something, but is this the only place this constant is used? If so then do we just want the default to be specified here?

File: official/keras_application_models/benchmark_main.py (hunk 3 above)
Comment (yhliang2018) [reply official/keras_application_models/benchmark_main.py]: Sure, I will add a warning message here. Then it keeps consistent with the help message in the --eager flag.

File: official/keras_application_models/benchmark_main.py (hunk 5 above)
Comment (yhliang2018) [reply official/keras_application_models/benchmark_main.py]: Yeah, we use 1000 as the default number of images for both train and eval with synthetic dataset. Will just put a value here.

**Comment by yhliang2018** [comment]
Hi Anjali and Karmel,

I have addressed all the comments. The benchmark is already set up. As long as this PR is approved, we should be able to get some benchmark numbers, like exp/sec.

Thanks,
Yanhui

File: official/keras_application_models/dataset.py [hunk 6]
```diff
@@ -19,6 +19,9 @@
 
 import tensorflow as tf
 
+# pylint: disable=g-bad-import-order
```
Comment (karmel) [inline official/keras_application_models/dataset.py]: This should be at the end of the line, or re-enabled after the relevant section

**Pull request merged**